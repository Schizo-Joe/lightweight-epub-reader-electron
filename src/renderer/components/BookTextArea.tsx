import * as React from 'react';
import {Book} from "epub-chinese-converter";

export interface Props {
    book?: Book.BookWithMeta
}

const BookTextArea = ({book}: Props) => {
    console.log(`at BookTextArea, book: ${book}`);

    if (book) {
        const chapters = book.chapters;
        return (
            <div>
                {Object.keys(chapters).map(chapterId => (
                    <div key={chapterId} dangerouslySetInnerHTML={{__html: chapters[chapterId].text}}/>
                ))}
            </div>
        );
    } else {
        return (
            <div>No book chosen yet</div>
        );
    }
};

export default BookTextArea;
