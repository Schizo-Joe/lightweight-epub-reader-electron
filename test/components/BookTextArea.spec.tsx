import * as React from 'react';
import * as renderer from 'react-test-renderer';

import BookTextArea from '../../src/renderer/components/BookTextArea';
import {Book} from "epub-chinese-converter";

describe('<BookTextArea/>', () => {
    it('renders to "No book chosen yet" when book not loaded correctly', () => {
        const tree = renderer
            .create(<BookTextArea book={undefined}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when book loaded', () => {
        // given
        const book: Book.BookWithMeta = {
            chapters: {
                ch1: {text: "this is chapter 1"},
                ch2: {text: "this is chapter 2"},
            }, metadata: {}
        };

        const tree = renderer
            .create(<BookTextArea book={book}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
