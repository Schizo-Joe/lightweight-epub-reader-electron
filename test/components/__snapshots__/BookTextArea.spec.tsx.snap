// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookTextArea/> renders correctly when book loaded 1`] = `
<div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "this is chapter 1",
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "this is chapter 2",
      }
    }
  />
</div>
`;

exports[`<BookTextArea/> renders to "No book chosen yet" when book not loaded correctly 1`] = `
<div>
  No book chosen yet
</div>
`;
